clippy_lints/src/doc/markdown.rs

```diff
@@ -92,13 +92,23 @@ fn check_word(cx: &LateContext<'_>, word: &str, span: Span, code_level: isize, b
     if let Ok(url) = Url::parse(word) {
         // try to get around the fact that `foo::bar` parses as a valid URL
         if !url.cannot_be_a_base() {
-            span_lint(
+            span_lint_and_then(
                 cx,
                 DOC_MARKDOWN,
                 span,
                 "you should put bare URLs between `<`/`>` or make a proper Markdown link",
+                |diag| {
+                    let mut applicability = Applicability::MachineApplicable;
+                    let snippet = snippet_with_applicability(cx, span, "..", &mut applicability);
+                    diag.span_suggestion_with_style(
+                        span,
+                        "put bare URLs between `<`/`>`",
```

Comment:
I don't like that this is basically repeating the message. How about something like
```suggestion
                        "try",
```
This is a fairly common one